protobuf/lib/src/protobuf/pb_map.dart

```diff
@@ -39,6 +39,35 @@ class PbMap<K, V> extends MapBase<K, V> {
     _wrappedMap[key] = value;
   }
 
+  @override
+  bool operator ==(other) {
+    if (identical(other, this)) {
+      return true;
+    }
+    if (other is! PbMap) {
+      return false;
+    }
+    if (other.length != length) {
+      return false;
+    }
+    if (other.hashCode != hashCode) {
+      return false;
+    }
+    for (final key in keys) {
+      if (other[key] != this[key]) {
```

Comment:
`other[key]` could be `null`.
I'd add a comment that `this[key]` can never be `null`, so it is not necessary to check `other.containsKey(key)`, since the test will fail for missing `key`.

As a TODO - It might be more efficient to check all the keys first, since they are usually smaller than the values.